vulnerabilities/scraper/archlinux.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2017 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/vulnerablecode/
+# The VulnerableCode software is licensed under the Apache License version 2.0.
+# Data generated with VulnerableCode require an acknowledgment.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import json
+from urllib.request import urlopen
+
+
+ARCHLINUX_TRACKER_URL = 'https://security.archlinux.org/json'
+
+
+def extract_vulnerabilities(arch_data):
+    """
+    Return a series of mappings for each record of arch linux security tracker
+    data.
+    """
+    for item in arch_data:
+        package_vulnerabilities = []
+        cves = item['issues']
+        advisories = set(item['advisories'])
+        vulnerabilities = cves + list(advisories)
+        vulnerabilities.append(item['name'])
+        packages_name = item['packages']
+
+        if not vulnerabilities or not packages_name:
+            continue
+
+        affected_version = item['affected']
+        fixed_version = item['fixed']
+        if not fixed_version:
+            fixed_version = 'None'
+
+        package_vulnerabilities = {
```

Comment:
@pombredanne I agree that `package_vulnerability` data structure  doesn't bring any direct value on top of the original data structure. But IMO its not good on part of users  to use the Arch JSON data directly to create models. I mean the users might want to view/observe the data before actually dumping it into the db. So, I'd rather suggest that instead of having  2 methods namely `extract_vulnerabilities` and `extract_vulnerabilities` we could have only 1 method as `extract_vulnerabilities` and leave `archlinux_dump` as it is. Please suggest your views on this.

Reply:
> I mean the users might want to view/observe the data before actually dumping it into the db. 

that's not the goal: we want to import fully automatically... the data will never ever be reviewed before it is imported

Reply:
so the only reason why you want to a tow step approach is for when the data is not structured in the first place. Here this is a JSON dump so there is not much of a transformation needed in a structure format.  The key is converting the data to proper DB records in the simplest way. Does this make sense to you?